src/main/java/org/zone/region/flag/player/entitydamage/EntityDamagePlayerFlagListener.java

```diff
@@ -0,0 +1,35 @@
+package org.zone.region.flag.player.entitydamage;
+
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.event.Listener;
+import org.spongepowered.api.event.entity.DamageEntityEvent;
+import org.zone.ZonePlugin;
+import org.zone.region.Zone;
+import org.zone.region.group.Group;
+import org.zone.region.group.key.GroupKeys;
+
+import java.util.Optional;
+
+public class EntityDamagePlayerFlagListener {
+    @Listener
+    public void onEntityDamageEvent(DamageEntityEvent event) {
+        if (!(event.entity() instanceof Player player)) {
+            return;
+        }
+
+        Optional<Zone> opZone = ZonePlugin
+                .getZonesPlugin()
+                .getZoneManager()
+                .getPriorityZone(event.entity().world(), event.entity().position());
+        if (opZone.isEmpty()) {
+            return;
+        }
+
+        Zone zone = opZone.get();
+        Group group = zone.getMembers().getGroup(player.uniqueId());
+        if (group.contains(GroupKeys.ENTITY_DAMAGE_PLAYER)) {
+            return;
+         }
```

Comment:
I got it wrong before and you were right. You should cancel the event if the player has the permission

Reply:
yes lol

Reply:
dats the reason i was weirded out lol